### PR TITLE
Reformat kotlin code

### PR DESCRIPTION
**Pull request by yufengwangca**
Add kotlin-format to format kotlin source code

### CODE REVIEW

File: scripts/helpers/restyle-diff.sh [hunk 1]
```diff
@@ -50,3 +50,9 @@ fi
 
 declare -a paths="($(git diff --ignore-submodules --name-only --merge-base "$ref"))"
 restyle-paths "${paths[@]}"
+
+if [ ! -f "/tmp/ktfmt-0.44-jar-with-dependencies.jar" ]; then
+    wget "https://repo1.maven.org/maven2/com/facebook/ktfmt/0.44/ktfmt-0.44-jar-with-dependencies.jar" -P /tmp
+fi
+
+find src examples -name '*.kt' | xargs java -jar /tmp/ktfmt-0.44-jar-with-dependencies.jar --google-style
```
Comment (bzbarsky-apple) [inline scripts/helpers/restyle-diff.sh:58]: This is not restyling the right set of files.  restyle-diff is supposed to restyle the files that differe from the base.

Also, this will break if "java" is not installed, even if nothing java-related is in the diff.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: scripts/helpers/restyle-diff.sh (hunk 1 above)
Comment (yufengwangca) [reply scripts/helpers/restyle-diff.sh]: Acked,  I separate kotlin format to a new shell script kotlin-format.sh, this script conducts the same reformat to conform the existing kotlin format check in CI

File: scripts/helpers/restyle-diff.sh [hunk 2]
```diff
@@ -48,5 +48,5 @@ if [[ -z "$ref" ]]; then
     git remote | grep -qxF upstream && ref="upstream/master"
 fi
 
-declare -a paths="($(git diff --ignore-submodules --name-only --merge-base "$ref"))"
+declare -a paths=("$(git diff --ignore-submodules --name-only --merge-base "$ref")")
```
Comment (bzbarsky-apple) [inline scripts/helpers/restyle-diff.sh:51]: Why the changes to this file?

**Review by bzbarsky-apple — APPROVED** [review]

File: scripts/helpers/restyle-diff.sh (hunk 2 above)
Comment (yufengwangca) [reply scripts/helpers/restyle-diff.sh:51]: They are equivalent, the old declaration uses the old syntax for assigning the output of a command to an array. The syntax is:
declare -a array_name="(command_output)"

The new declaration uses the new syntax for assigning the output of a command to an array. The syntax is:
declare -a array_name=(command_output)

The new syntax is more concise and easier to read, so it is the preferred syntax.

Comment (bzbarsky-apple) [reply scripts/helpers/restyle-diff.sh:51]: Yes, but why are you touching that file at all?

**Review by andy31415 — APPROVED** [review]
